asf_welsh_energy_consultation/pipeline/process_data.py

```diff
@@ -114,14 +114,38 @@ def cumsums_by_variable(variable, new_var_name, data=enhanced_mcs):
 wales_epc = get_data.get_wales_processed_epc()
 
 
-def get_wales_epc_new():
+def correct_new_dwelling_labels():
+    """
+    For each unique property in the pandas DataFrame that has more than one record where `TRANSACTION_TYPE == "new dwelling"`,
+    replace "new dwelling" with "unknown" for each row except the row with the earliest date.
+
+    Returns:
+        pd.DataFrame: Wales EPC certificates
+    """
+    wales_epc["rank"] = wales_epc.groupby("BUILDING_REFERENCE_NUMBER")[
+        "INSPECTION_DATE"
+    ].rank("dense")
```

Comment:
i think we need to use UPRN here, as 1 building might have multiple flats. so if we use `BUILDING_REFERENCE_NUMBER` the results will be wrong
```suggestion
    wales_epc["rank"] = wales_epc.groupby("UPRN")[
        "INSPECTION_DATE"
    ].rank("dense")
```

Reply:
Great point thank you! Got them mixed up here.